src/movies.js

```diff
@@ -1,25 +1,175 @@
+  
+
+function howManyMovies(moviesArray) {
+  let result = 0
+  if (moviesArray.length===0){
+    return result
+  }
+
+  let moviesSpielberg = moviesArray.filter (movie => movie.director===`Steven Spielberg`)
+  let onlyDrama =[]
+  for ( let i = 0; i < moviesSpielberg.length; i ++){
+    let genre = moviesSpielberg[i].genre
+
+    for ( let j = 0; j < genre.length; j++ ) {
+      const element = genre[j];
+      
+      if (element==="Drama") {
+        result++
+        onlyDrama.push(moviesSpielberg[i]) 
+      }
+      
+    }
+    
+  }
+  
+
+ return result
+
+}
+
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+
+function scoresAverage(moviesArray) {
+  let result = 0
+  if (moviesArray.length === 0) {//if is empty = 0
+    return result
+  }
+
+ let scores= moviesArray.map(scores=> scores.score) 
+ let suma = 0
+ for (let i = 0 ; i < scores.length; i ++){
+  let element = scores[i];
+  if (element === " " || element === "" || element === undefined ) {//if if one of the movies does not have score = 0
+    element = 0
+    
+  }
+  suma += element 
+  
+ }
+
+let avg = suma / scores.length// average 
+result = avg
+
+ return parseFloat(result.toFixed(2))//number with two decimals
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+console.log("hola")
+
+
+function dramaMoviesScore(moviesArray) {
+  
+  const dramaMovies = moviesArray.filter(movie => movie.genre.includes('Drama'));console.log(typeof dramaMovies)//creamos un array con las pelis de drama
+  if (dramaMovies.length === 0) {
+    return 0
+  }
+  let average=scoresAverage(dramaMovies)
+  return average
+  }
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear(moviesArray) {}
+function orderByYear(moviesArray) {
+
+
+  const copyMoviesArray=moviesArray.slice()//hacemos una copia del array para que no modifique el original
+  
+  copyMoviesArray.sort((a,b)=>{//primero ordenamos alfabeticamente
+    if ( a.title < b.title){
+      return -1
+    }
+    if ( a.title > b.title ){
+      return 1
+    }
+    return 0
+  })
+
+
+  copyMoviesArray.sort((a,b)=> a.year - b.year)//despues ordenamos por año
+  
+
+ return copyMoviesArray
+
+}
 
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
-function orderAlphabetically(moviesArray) {}
+function orderAlphabetically(moviesArray) {
+  
+  
+    let onlyTwenty=moviesArray.slice()//hacemos una copia del array para que no modifique el original
+    
+    onlyTwenty=onlyTwenty.map (movie=>movie.title)//ahora hacemos un map para quedarnos cons titles
+     onlyTwenty.sort ((a, b) => a.localeCompare(b));//los ordenamos alfabeticamente
+     if (onlyTwenty.length >= 20){//si el leght de onlyTwenty es igual o mayor a 20 me devuelves los 20 primeros
+  
+      onlyTwenty= onlyTwenty.slice(0,20)
+    }
+
+  return onlyTwenty
+  }
```

Comment:
Vamos a refactorizar esto!

```suggestion
function orderAlphabetically(moviesArray) {
  return moviesArray.map(movie=>movie.title).sort((a, b) => a.localeCompare(b)).slice(0,20)
}
```

El metodo map te devuelve un array nuevo asi que perfecto! ya tenemos el array nuevo, fijate que digo que es un array, pues tenemos acceso a sus metodos, como el sort, lo usamos y para final como tenemos que coger los 20 primeros pues el slice de 20 y listo!!!